mkosi/machine.py

```diff
@@ -0,0 +1,129 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+from __future__ import annotations
+
+import contextlib
+import enum
+import signal
+import subprocess
+import sys
+from typing import Any, List, Optional, Sequence
+
+import pexpect  # type: ignore
+
+from . import (
+    MKOSI_COMMANDS_SUDO,
+    build_stuff,
+    bump_image_version,
+    check_native,
+    check_output,
+    check_root,
+    init_namespace,
+    load_args,
+    needs_build,
+    parse_args,
+    prepend_to_environ_path,
+    print_output_size,
+    run_qemu_cmdline,
+    run_shell_cmdline,
+    save_manifest,
+    suppress_stacktrace,
+    unlink_output,
+    virt_name,
+)
+from .backend import die, run
+
+
+class Machine:
+    def __init__(self, args: Optional[Sequence[str]] = None) -> None:
+        # Remains None until image is built and booted, then receives pexpect process
+        self._serial: Optional[pexpect.spawnu] = None
```

Comment:
Type here is wrong I think, the function is called `spawnu` but the object returned by `spawnu` is `pexpect.spawn`. If you fix this you might also be able to remove the `type: ignore` above.

Reply:
Ok, there's nothing we can do about the mypy error so keeping the type: ignore is fine